packages/server/lib/browsers/cdp_automation.ts

```diff
@@ -174,18 +176,36 @@ const ffToStandardResourceTypeMap: { [ff: string]: ResourceType } = {
   'webmanifest': 'manifest',
 }
 
+export interface CdpOptions {
+  sendDebuggerCommandFn: SendDebuggerCommand
+  onFn: OnFn
+  automation: Automation
+  experimentalSessionAndOrigin: boolean
+}
+
 export class CdpAutomation {
-  constructor (private sendDebuggerCommandFn: SendDebuggerCommand, onFn: OnFn, private automation: Automation) {
-    onFn('Network.requestWillBeSent', this.onNetworkRequestWillBeSent)
-    onFn('Network.responseReceived', this.onResponseReceived)
-    sendDebuggerCommandFn('Network.enable', {
+  sendDebuggerCommandFn: SendDebuggerCommand
+  automation: Automation
+  experimentalSessionAndOrigin: boolean
+
+  constructor (options: CdpOptions) {
+    this.sendDebuggerCommandFn = options.sendDebuggerCommandFn
+    this.automation = options.automation
+    this.experimentalSessionAndOrigin = options.experimentalSessionAndOrigin
+
+    options.onFn('Network.requestWillBeSent', this.onNetworkRequestWillBeSent)
+    options.onFn('Network.responseReceived', this.onResponseReceived)
+  }
+
+  async enable () {
+    await this.sendDebuggerCommandFn('Network.enable', {
```

Comment:
Yay for moving this out of the constructor since it's a promise. You might have some conflicts when you go to 10 though :)

Reply:
Yeah, not really looking forward to that merge 😅